code/game/machinery/doors/windowdoor.dm

```diff
@@ -290,19 +297,53 @@
 			AE.conf_access = src.req_one_access
 			AE.one_access = 1
 
-	return WA
-
 /obj/machinery/door/window/brigdoor
-	name = "Secure Door"
+	name = "Secure Window Door"
 	icon = 'icons/obj/doors/windoor.dmi'
 	icon_state = "leftsecure"
 	base_state = "leftsecure"
 	req_access = list(access_security)
+	secure = 1
 	var/id_tag = null
-	health = 300.0 //Stronger doors for prison (regular window door health is 200)
+	health = 200
 
 /obj/machinery/door/window/brigdoor/make_assembly(mob/user as mob)
 	var/obj/structure/windoor_assembly/WA = ..(user)
 	WA.secure = "secure_"
 	WA.update_icon()
 	return WA
+
+/obj/machinery/door/window/plasma
+	name = "Plasma Window Door"
+	desc = "A sliding glass door strengthened by plasma."
+	icon = 'icons/obj/doors/plasmawindoor.dmi'
+	health = 300
+
+/obj/machinery/door/window/plasma/make_assembly(mob/user as mob)
+	// Windoor assembly
+	var/obj/structure/windoor_assembly/plasma/WA = new /obj/structure/windoor_assembly/plasma(src.loc)
+	set_assembly(user, WA)
+	return WA
+
+/obj/machinery/door/window/plasma/take_damage(var/damage)
+	src.health = max(0, src.health - damage)
+	if (src.health <= 0)
+		getFromPool(/obj/item/weapon/shard/plasma, loc)
+		var/obj/item/stack/cable_coil/CC = new /obj/item/stack/cable_coil(src.loc)

```

Comment:
Yeah that's true that proc does pretty much just copy the other, wouldn't say an 'exceptional' amount of copied code though